migrations/1601022564938-copy-parents-data-into-children-entities-and-delete-parent-users.js

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable no-await-in-loop */
+// eslint-disable-next-line no-unused-vars
+const mongoose = require('mongoose');
+
+const { info } = require('../src/logger');
+
+const { connect, close } = require('../src/utils/database');
+
+const parentSchema = new mongoose.Schema({
+	firstName: { type: String, required: true },
+	lastName: { type: String, required: true },
+	email: { type: String, required: true, lowercase: true },
+});
+
+const consentForm = ['analog', 'digital', 'update'];
+
+const OldUser = mongoose.model(
+	'oldUser',
+	new mongoose.Schema({
+		parents: [{ type: mongoose.Schema.Types.ObjectId, ref: 'user' }],
+		consent: {
+			parentConsents: [
+				{
+					parentId: { type: mongoose.Schema.Types.ObjectId, ref: 'user' },
+					form: { type: String, enum: consentForm },
+					dateOfPrivacyConsent: { type: Date },
+					dateOfTermsOfUseConsent: { type: Date },
+					privacyConsent: { type: Boolean },
+					termsOfUseConsent: { type: Boolean },
+				},
+			],
+		},
+	}),
+	'users'
+);
+
+const User = mongoose.model(
+	'myUserModel',
+	new mongoose.Schema({
+		roles: [{ type: mongoose.Schema.Types.ObjectId, ref: 'role' }],
+		parents: [parentSchema],
+		consent: {
+			parentConsents: [
+				{
+					parentId: { type: mongoose.Schema.Types.ObjectId, ref: 'user' },
+					form: { type: String, enum: consentForm },
+					dateOfPrivacyConsent: { type: Date },
+					dateOfTermsOfUseConsent: { type: Date },
+					privacyConsent: { type: Boolean },
+					termsOfUseConsent: { type: Boolean },
+				},
+			],
+		},
+	}),
+	'users'
+);
+
+const getIds = (doc) =>
+	doc.reduce((prev, curr) => {
+		prev.push(curr._id);
+		return prev;
+	}, []);
+
+const findAllParents = async () =>
+	User.aggregate([
+		{
+			$lookup: {
+				from: 'roles',
+				localField: 'roles',
+				foreignField: '_id',
+				as: 'rolesLookup',
+			},
+		},
+		{ $match: { 'rolesLookup.name': 'parent' } },
+	]);
+
+const deleteParentUsers = (parents) => {
+	const idsToDelete = getIds(parents);
+	info(`${idsToDelete.length} parent users will be deleted!`);
+	return User.deleteMany({ _id: { $in: [idsToDelete] } });
+};
```

Comment:
```suggestion
const parentRole = await RoleModel.findOne({ name: 'parent' }).lean().exec();

const findAllParents = async () => User.find({ role: parentRole._id }).lean().exec();

const deleteParentUsers = async () => {
	const result = await User.deleteMany({ role: parentRole._id }).exec();
	info(`${result.deletedCount} users deleted.`);
};
```
I think this is much cleaner (and probably a lot faster), despite one additional query.

Reply:
done